compiler/src/dotty/tools/repl/ReplDriver.scala

```diff
@@ -209,21 +227,51 @@ class ReplDriver(settings: Array[String],
     compiler
       .compile(parsed)
       .fold(
-        displayErrors,
+        {
+          case (errors: Errors) =>
+            if (!silent) displayErrors(errors)
+            state
+        },
         {
           case (unit: CompilationUnit, newState: State) => {
             val newestWrapper = extractNewestWrapper(unit.untpdTree)
             val newImports = newState.imports ++ extractImports(parsed.trees)(newState.run.runContext)
             val newStateWithImports = newState.copy(imports = newImports)
 
-            displayDefinitions(unit.tpdTree, newestWrapper)(newStateWithImports)
+            val (decls, optIndex, symbol) = getDefinitions(unit.tpdTree, newestWrapper)(newStateWithImports)
+            if (!silent) displayDefinitions(decls, symbol)(newStateWithImports)
+
+            optIndex.map(i => newStateWithImports.copy(valIndex = i)).getOrElse(newStateWithImports)
           }
         }
       )
   }
 
-  /** Display definitions from `tree` */
-  private def displayDefinitions(tree: tpd.Tree, newestWrapper: Name)(implicit state: State): State = {
+  /** Display definitions from `symbol` */
+  private def displayDefinitions(decls: Seq[String], symbol: Option[Symbol])(implicit state: State): Unit = {
+    implicit val ctx = state.run.runContext
+
+    def displayMembers(decls: Seq[String]): Unit = {
+      decls.foreach(str => out.println(SyntaxHighlighting(str)))
+    }
+
+    def isSyntheticCompanion(sym: Symbol) =
+      sym.is(Module) && sym.is(Synthetic)
```

Comment:
`sym.is(Module | Synthetic)`

Reply:
Changing `sym.is(Module) && sym.is(Synthetic)` to `sym.is(Module | Synthetic)` made some test cases failed. So I didn't use this as of now

Reply:
This should not be the case. Could you try again with `sym.is(Module | Synthetic)`. If that does not work you might have found a bug.

Reply:
I tried again. This is the error
```
scala> Test.foo(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109)
val res0: Int = 42
actual ===========>
scala> object Test { def foo(p1: Int, p2: Int, p3: Int, p4: Int, p5: Int, p6: Int, p7: Int, p8: Int, p9: Int, p10: Int, p11: Int, p12: Int, p13: Int, p14: Int, p15: Int, p16: Int, p17: Int, p18: Int, p19: Int, p20: Int, p21: Int, p22: Int, p23: Int, p24: Int, p25: Int, p26: Int, p27: Int, p28: Int, p29: Int, p30: Int, p31: Int, p32: Int, p33: Int, p34: Int, p35: Int, p36: Int, p37: Int, p38: Int, p39: Int, p40: Int, p41: Int, p42: Int, p43: Int, p44: Int, p45: Int, p46: Int, p47: Int, p48: Int, p49: Int, p50: Int, p51: Int, p52: Int, p53: Int, p54: Int, p55: Int, p56: Int, p57: Int, p58: Int, p59: Int, p60: Int, p61: Int, p62: Int, p63: Int, p64: Int, p65: Int, p66: Int, p67: Int, p68: Int, p69: Int, p70: Int, p71: Int, p72: Int, p73: Int, p74: Int, p75: Int, p76: Int, p77: Int, p78: Int, p79: Int, p80: Int, p81: Int, p82: Int, p83: Int, p84: Int, p85: Int, p86: Int, p87: Int, p88: Int, p89: Int, p90: Int, p91: Int, p92: Int, p93: Int, p94: Int, p95: Int, p96: Int, p97: Int, p98: Int, p99: Int, p100: Int, p101: Int, p102: Int, p103: Int, p104: Int, p105: Int, p106: Int, p107: Int, p108: Int, p109: Int): Int = 42 }
scala> Test.foo(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109)
val res0: Int = 42
[error] Test dotty.tools.repl.ScriptedTests.replTests failed: java.lang.AssertionError: Error in file /Users/rveeranki/code/dotty/compiler/target/scala-2.12/test-classes/repl/functions, expected output did not match actual, took 1.259 sec
```
Basically `Object Foo` doesn't print anything.

Reply:
It looks like you're not printing the definition of `Foo` for some reason, right?

Have a look at: https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/repl/ReplDriver.scala#L226

Reply:
I even tried on current master by changing `sym.is(Module) && sym.is(Synthetic)` to `sym.is(Module | Synthetic)`
Repl goes silent for `object Foo` however if I change it to `sym.is(Module & Synthetic)` it prints `// defined object foo`. So it looks like this behaviour is not introduced by these changes.